clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -1057,6 +1074,23 @@ public void recordTopicFetchMetrics(String topic, int bytes, int records) {
             }
             recordsFetched.record(records);
         }
+
+        public void recordPartitionFetchLag(TopicPartition tp, long lag) {
+            String name = tp + ".records-lag";
+            Sensor recordsLag = this.metrics.getSensor(name);
+            if (recordsLag == null) {
+                recordsLag = this.metrics.sensor(name);
+                recordsLag.add(this.metrics.metricName(name, this.metricGrpName, "The latest lag of the partition"),
+                               new Value());
+                recordsLag.add(this.metrics.metricName(tp + ".records-lag-max",
+                        this.metricGrpName,
+                        "The max lag of the partition"), new Max());
+                recordsLag.add(this.metrics.metricName(tp + ".records-lag-avg",
```

Comment:
I was going to leave this, but since there's one other change to be made, I think you can just do:

```java
recordsLag.add(this.metrics.metricName(name + "-max",
```

And similarly for `avg`. That avoids having to compute `toString` on the topic partition again.